components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/services/EventTrackerService.kt

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package io.gitpod.jetbrains.remote.services
+
+import com.intellij.codeWithMe.ClientId
+import com.intellij.openapi.application.ApplicationInfo
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.openapi.diagnostic.thisLogger
+import io.gitpod.gitpodprotocol.api.entities.RemoteTrackMessage
+import io.gitpod.jetbrains.remote.GitpodManager
+import io.gitpod.supervisor.api.Info.WorkspaceInfoResponse
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.future.await
+import kotlinx.coroutines.launch
+
+@Service
+class EventTrackerService {
+
+    private val manager = service<GitpodManager>()
+    private lateinit var info: WorkspaceInfoResponse
+    private val versionName = ApplicationInfo.getInstance().versionName
+    private val fullVersion = ApplicationInfo.getInstance().fullVersion
+
+    init {
+        GlobalScope.launch {
+            info = manager.pendingInfo.await()
+        }
+    }
+
+    fun trackEvent(eventName: String, props: Map<String, Any?>) {
```

Comment:
it seems to be duplication of GitpodClientProjectSessionTracker. Why cannot we implement this hook there? In order to a editor be focused there should be a project and associated window. I would imagine it should be possible.

Reply:
i.e with something like:
```
project.messageBus.connect(this)
      .subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, this)
```

in GitpodClientProjectSessionTracker?

Reply:
What I was thinking was to make `track` a `service` and then we can track anywhere easier, so I plan to change `jb_session` to use this track service too

Reply:
I think we introduced GitpodClientProjectSessionTracker on purpose as a project service to ensure that we don't track events from backend itself (It is own of clients) and only track for properly registered clients. Would it be preserved with a new approach? I think it is more safe to use a session passed in GitpodClientProjectSessionTracker.